### PR TITLE
Prompt to override existing LFS filters

### LINKED ISSUE #2693: Error in newest beta build: The filter.lfs.clean attribute should be "git-lfs clean -- %f"
### Description

My Github desktop beta just updated to the latest release, and I'm now getting this error on launch:

The filter.lfs.clean attribute should be "git-lfs clean -- %f" but is "git lfs clean %f"
Run `git lfs install --force` to reset git config.

### Version

**GitHub Desktop version:** 0.8.1

**OS version:** macOS 10.13 Beta (17A360a)

I also use homebrew and have git 2.14.1 installed via brew.

### PR DESCRIPTION
**Pull request by joshaber**
Fixes #2693 

screen shot 2017-09-13 at 10 43 45 am

### CODE REVIEW

**Comment by joshaber** [comment]
This is ready 🎡

**Comment by iAmWillShepherd** [comment]
I think it would be nice to show or have an option to show what the values will be updated to...

**Comment by joshaber** [comment]
Yeah it would, though currently Dugite doesn't include captures in the errors it reports. I'll open a more general issue about that in https://github.com/desktop/dugite, but I don't think it needs to hold this up.

**Comment by joshaber** [comment]
Opened https://github.com/desktop/dugite/issues/124 to track that 💥

File: app/src/ui/lfs/attribute-mismatch.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  readonly onDismissed: () => void
+
+  /** Called when the user has chosen to replace the update filters. */
+  readonly onUpdateExistingFilters: () => void
+}
+
+export class AttributeMismatch extends React.Component<
+  IAttributeMismatchProps,
+  {}
+> {
```
Comment (iAmWillShepherd) [inline app/src/ui/lfs/attribute-mismatch.tsx]: I know this isn't you, but seriously?! This ain't pretty 😞

File: app/src/ui/lfs/attribute-mismatch.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  public render() {
+    return (
+      <Dialog
+        id="lfs-attribute-mismatch"
+        title={
+          __DARWIN__ ? (
```
Comment (iAmWillShepherd) [inline app/src/ui/lfs/attribute-mismatch.tsx]: Why the parenthesis here?

File: app/src/ui/lfs/attribute-mismatch.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          ) : (
+            'Update existing Git LFS filters?'
+          )
+        }
+        onDismissed={this.props.onDismissed}
+        onSubmit={this.props.onUpdateExistingFilters}
+      >
+        <DialogContent>
+          <p>
+            Git LFS filters are already configured in your global git config but
```
Comment (iAmWillShepherd) [inline app/src/ui/lfs/attribute-mismatch.tsx]: Could we make _global git config_ open the file when clicked? That way people don't have to leave the app to check it out.

File: app/src/ui/lfs/attribute-mismatch.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            are not the values it expects. Would you like to update them now?
+          </p>
+        </DialogContent>
+
+        <DialogFooter>
+          <ButtonGroup>
+            <Button type="submit">
+              {__DARWIN__ ? (
+                'Update Existing Filters'
```
Comment (iAmWillShepherd) [inline app/src/ui/lfs/attribute-mismatch.tsx]: Same, why extra parenthesis?

File: app/src/ui/lfs/attribute-mismatch.tsx (hunk 2 above)
Comment (joshaber) [reply app/src/ui/lfs/attribute-mismatch.tsx]: That's all Prettier's doing 🙄

File: app/src/ui/lfs/attribute-mismatch.tsx (hunk 4 above)
Comment (joshaber) [reply app/src/ui/lfs/attribute-mismatch.tsx]: Same, all Prettier ➿

File: app/src/ui/lfs/attribute-mismatch.tsx (hunk 2 above)
Comment (iAmWillShepherd) [reply app/src/ui/lfs/attribute-mismatch.tsx]: I'm starting to think prettier ~doesn't match my opinions~ is too opinionated.

Comment (j-f1) [reply app/src/ui/lfs/attribute-mismatch.tsx]: @iAmWillShepherd Open an issue on prettier/prettier then 😄

Comment (iAmWillShepherd) [reply app/src/ui/lfs/attribute-mismatch.tsx]: I can see that going over well! lol Issue title: **I disagree with your opinions, my opinions are more correct than your opinions, so change your opinions to be in sync with my opinions.** I'll adapt.

**Comment by joshaber** [comment]
🍘

File: app/src/ui/lfs/attribute-mismatch.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    } catch (error) {
+      log.warn(`Couldn't get the global git config path`, error)
+    }
+  }
+
+  private renderGlobalGitConfigLink() {
+    const path = this.state.globalGitConfigPath
+    const msg = 'your global git config'
+    if (path) {
+      return <LinkButton onClick={this.showGlobalGitConfig}>{msg}</LinkButton>
```
Comment (iAmWillShepherd) [inline app/src/ui/lfs/attribute-mismatch.tsx]: We should only render this as a link if the path exists

**Pull request merged**